cli/cmdlineparser.cpp

```diff
@@ -1131,6 +1148,18 @@ CmdLineParser::Result CmdLineParser::parseFromArgs(int argc, const char* const a
                             return Result::Fail;
                         }
 
+                        if (rule.severity == Severity::none) {
+                            mLogger.printError("unable to load rule-file '" + ruleFile + "' - a rule has an invalid severity.");
+                            return Result::Fail;
+                        }
+
+                        rule.regex = std::make_shared<Regex>(rule.pattern);
+                        const std::string regex_err = rule.regex->compile();
```

Comment:
I am not too happy about this pattern but I could not come up with some without changing the code quite a bit which I wanted to avoid.